src/agent/rustjail/src/container.rs

```diff
@@ -1560,3 +1642,26 @@ fn execute_hook(logger: &Logger, h: &Hook, st: &OCIState) -> Result<()> {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_status_transtition() {
+        let mut status = ContainerStatus::new();
+        let status_table: [Status; 4] = [
+            Status::CREATED,
```

Comment:
`Status::UNKNOWN` isn't being tested. But see above as maybe we can remove this state?

Reply:
Hi @jodh-intel 

All of the comments are addressed.